test/AutoRest.TestServerLowLevel.Tests/dpg-customization.cs

```diff
@@ -74,6 +74,61 @@ await foreach (Page<BinaryData> page in allPages.AsPages())
             }
         });
 
+        [Test]
+        public Task HandwrittenModelLro() => Test(async (host) =>
+        {
+            using var diagnosticListener = new ClientDiagnosticListener("dpg_customization_LowLevel", asyncLocal: true);
+            CollectionAssert.IsEmpty(diagnosticListener.Scopes);
+
+            Operation<Product> lro = await new DPGClient(Key, host).LroValueAsync(WaitUntil.Started, "model");
+            Assert.AreEqual(1, diagnosticListener.Scopes.Count);
+            Assert.AreEqual("DPGClient.LroValue", diagnosticListener.Scopes[0].Name);
+            Assert.IsTrue(diagnosticListener.Scopes[0].IsCompleted);
+
+            await lro.WaitForCompletionAsync();
+            Assert.AreEqual(2, diagnosticListener.Scopes.Count);
+            Assert.AreEqual("model", $"{lro.Value.Received}");
+            Assert.AreEqual("DPGClient.LroValue", diagnosticListener.Scopes[1].Name);
```

Comment:
The name of the public method call is `WaitForCompletion` ... shouldn't that be the name of the scope?

Reply:
Ok